chain/repository.go

```diff
@@ -322,23 +322,34 @@ func (r *Repository) GetMaxBlockNum() (uint32, error) {
 
 // GetBlockSummary get block summary by block id.
 func (r *Repository) GetBlockSummary(id thor.Bytes32) (summary *BlockSummary, err error) {
-	var cached interface{}
-	if cached, err = r.caches.summaries.GetOrLoad(id, func() (interface{}, error) {
+	var blk interface{}
+	var cached bool
+	if blk, cached, err = r.caches.summaries.GetOrLoad(id, func() (interface{}, error) {
```

Comment:
Given the context is available we can probably do something like
```suggestion
	var blk interface{}
	cached := true
	if blk, err = r.caches.summaries.GetOrLoad(id, func() (interface{}, error) {
	cached = false
```